tripal_chado/tests/src/Kernel/Plugin/TripalBackendPublish/ChadoPublishTest.php

```diff
@@ -334,6 +335,26 @@ public function testTripalPublishServiceSingleJob() {
       $this->assertEquals($expected, $this->countFieldTable($field_table),
         'The drupal field table does not contain the expected number of publications for the analysis');
     }
+    // Tests unpublishing.
+    // Try to unpublish orphaned, but since there are no orphaned entities
+    // this should do nothing.
+    $publish_options = ['bundle' => 'organism', 'datastore' => 'chado_storage', 'schema_name' => $this->testSchemaName, 'unpublish' => TRUE];
+    $unpublished_entities = $this->chado_publish->publish($publish_options);
+    $this->assertCount(0, $unpublished_entities, 'No orphans, should not have unpublished any records');
+
+    // Delete one chado record and confirm that we can now unpublish it as
+    // an orphaned record.
+    $n = $this->connection->delete('1:organism')
+      ->condition('organism_id', 2, '=')
+      ->execute();
+    $this->assertEquals(1, $n, 'Did not delete organism from chado where organism_id=2');
+    $unpublished_entities = $this->chado_publish->publish($publish_options);
+    $this->assertCount(1, $unpublished_entities, 'Did not unpublish one orphaned organism entity');
+
+    // Unpublish all remaining non-orphaned organism entities.
```

Comment:
This is the one we really want to check that the chado records remain untouched for! As well as confirming that there are no entities for this content type anymore and also that field records for those entities were deleted as well.